cmd/jaeger/internal/exporters/storageexporter/exporter_test.go

```diff
@@ -0,0 +1,103 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storageexporter
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal/extension/jaegerstorage"
+	memoryCfg "github.com/jaegertracing/jaeger/pkg/memory/config"
+)
+
+var (
+	storage_exporter *storageExporter
+	storage_host     storageHost
+)
+
+type storageHost struct {
+	component *component.Component
+}
+
+func (host storageHost) GetExtensions() map[component.ID]component.Component {
+	myMap := make(map[component.ID]component.Component)
+	myMap[component.NewID("jaeger_storage")] = *host.component
+	return myMap
+}
+
+func (storageHost) ReportFatalError(err error) {
+	fmt.Println(err)
+}
+
+func (storageHost) GetFactory(_ component.Kind, _ component.Type) component.Factory {
+	return nil
+}
+
+func (storageHost) GetExporters() map[component.DataType]map[component.ID]component.Component {
+	return nil
+}
+
+const memstoreName = "memstore"
+
+func TestExporter(t *testing.T) {
+	config := &Config{}
+	config.TraceStorage = memstoreName
+	telemetry_settings := component.TelemetrySettings{
+		Logger: zap.L(),
+	}
+	storage_exporter = newExporter(config, telemetry_settings)
+	assert.Equal(t, storage_exporter.logger, telemetry_settings.Logger)
+	assert.Equal(t, storage_exporter.config, config)
+
+	storage_factory := jaegerstorage.NewFactory()
+	storage_config := jaegerstorage.Config{Memory: map[string]memoryCfg.Configuration{
+		memstoreName: {MaxTraces: 10000},
+	}}
+	storage_config.Memory[memstoreName] = memoryCfg.Configuration{MaxTraces: 10000}
+	storage_component, _ := storage_factory.CreateExtension(
+		context.Background(),
+		extension.CreateSettings{
+			TelemetrySettings: telemetry_settings,
+		},
+		&storage_config)
+	storage_host = storageHost{
+		component: &storage_component,
+	}
+
+	storage_component.Start(context.Background(), storage_host)
+	err := storage_exporter.start(context.Background(), storage_host)
+	assert.NotNil(t, storage_exporter.spanWriter)
+	require.NoError(t, err)
+
+	traces := ptrace.NewTraces()
+	rSpans := traces.ResourceSpans().AppendEmpty()
+	sSpans := rSpans.ScopeSpans().AppendEmpty()
+	span := sSpans.Spans().AppendEmpty()
+	span.SetName("test")
+
+	err = storage_exporter.pushTraces(context.Background(), traces)
+	require.NoError(t, err)
```

Comment:
aside from no error, you can also give a specific ID to the span and then validate that the memstore received it.

Reply:
I did that but I cannot figure out how to validate whether the memstore received it. Basically I need a way to check the traces in the memstore. I cannot find a way to do that.

Reply:
you can do something like storageExtension.GetStorageFactory().CreateSpanReader().GetTrace()